troposphere/static/js/modals/instance/resume.js

```diff
@@ -1,16 +1,31 @@
 import ModalHelpers from "components/modals/ModalHelpers";
-
 import InstanceResumeModal from "components/modals/instance/InstanceResumeModal";
+
 import actions from "actions";
 
 
 export default {
+    /**
+     * Resume any valid/actionable resources
+     *
+     * The action method handles resources verifying type (via instanceof)
+     * to ensure that multi0ple resources can be handled.
+     *
+     * @param resources - collection of resources to act on
+     */
+    resume: function(resources) {
+        // `resources` is a reference to the "selected" resources we are
+        // acting on; it's `props` - we know that it could be mutated by
+        // the owner - so we are going to clone this to avoid it changing
+        // out from under the modal/action calls
+        //
+        // ^ if you think this sounds totally wrong, track down @lenards
+        //   and set me straight ...
+        let resourcesCopy = resources.clone(),
```

Comment:
I actually think that there is not culprit, in the _unselect_ handler ... 
```
    onResourceDeselected: function(resource) {
        var selectedResources = this.state.selectedResources,
            previewedResource = this.state.previewedResource;

        // Remove the resources from the list of selected resources
        selectedResources.remove(resource);

        // Replace preview, with another
        if (previewedResource == resource) {
            previewedResource = selectedResources.last();
        }

        this.setState({
            previewedResource,
            selectedResources,
        });
    },
```
source: [ProjectDetails.jsx#L46-L62](https://github.com/lenards/troposphere/blob/0ae147f3dabd06b997eb802aced23590b6b0c1b7/troposphere/static/js/components/projects/detail/resources/ProjectDetails.jsx#L46-L62)

I attempted to map this function over the selections to avoid including "onUnselectAll", and and when failed, I shifted to the "clone()" as protection from having the reference mutated.